wandb/sdk/wandb_run.py

```diff
@@ -335,6 +339,24 @@ def wrapper(self: Type["Run"], *args: Any, **kwargs: Any) -> Any:
 
         return wrapper
 
+    @classmethod
+    def _noop_on_finish(cls, message: str = "") -> Callable:
+        def decorator_fn(func: Callable) -> Callable:
+            @functools.wraps(func)
+            def wrapper_fn(self: Type["Run"], *args: Any, **kwargs: Any) -> Any:
+                if not getattr(self, "_is_finished", False):
+                    return func(self, *args, **kwargs)
+
+                default_message = (
+                    f"Run ({self.id}) is finished. The call to `{func.__name__}` will be ignored. "
+                    f"Please make sure that you are using an active run."
+                )
+                warnings.warn(message or default_message, UserWarning, stacklevel=2)
```

Comment:
make this error for methods the user calls explicitly like `log` and warning for things like console logging since we are doing it on behalf of the user